modules/webxr/native/library_godot_webxr.js

```diff
@@ -514,18 +350,17 @@ const GodotWebXR = {
 		const viewport = glLayer.getViewport(view);
 		const gl = GodotWebXR.gl;
 
-		const orig_framebuffer = gl.getParameter(gl.FRAMEBUFFER_BINDING);
-		const orig_viewport = gl.getParameter(gl.VIEWPORT);
-
-		// Bind to WebXR's framebuffer.
-		gl.bindFramebuffer(gl.FRAMEBUFFER, glLayer.framebuffer);
-		gl.viewport(viewport.x, viewport.y, viewport.width, viewport.height);
+		const framebuffer = GL.framebuffers[p_destination_fbo];
 
-		GodotWebXR.blitTexture(gl, GodotWebXR.textures[view_index]);
+		gl.bindFramebuffer(gl.FRAMEBUFFER, null);
+		gl.bindFramebuffer(gl.READ_FRAMEBUFFER, framebuffer);
+		gl.readBuffer(gl.COLOR_ATTACHMENT0);
+		gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, glLayer.framebuffer);
 
-		// Restore state.
-		gl.bindFramebuffer(gl.FRAMEBUFFER, orig_framebuffer);
-		gl.viewport(orig_viewport[0], orig_viewport[1], orig_viewport[2], orig_viewport[3]);
+		// Flip Y upside down on destination.
+		gl.blitFramebuffer(0, 0, viewport.width, viewport.height,
+			viewport.x, viewport.height, viewport.width, viewport.y,
+			gl.COLOR_BUFFER_BIT, gl.NEAREST);
```

Comment:
Is there a reason state isn't being restored after this as it used to? OpenGL is really particularly about having state set correctly.

Reply:
No, not really :-) I think I was copying this code from `RasterizerGLES3::_blit_render_target_to_screen()` which is called at about the same time, and it doesn't reset the framebuffer bindings.

I'll update this to save and restore the state, similar to how the older code did it.

Thanks for catching this!

Reply:
Alright, I've added code that saves and restores the state, similar to the older code. It's still working on real devices in my testing!